feat(breadcrumb): draft of breadcrumb web component

mndonx opened the pull request:
## Summary

This PR adds a Breadcrumb menu web component to the UI Kit.

<img width="593" alt="Screenshot 2023-11-19 at 10 51 08 PM" src="https://github.com/emulsify-ds/emulsify-web-components/assets/2343957/740fa5a4-3499-41b4-a1e4-eb8e98ff1dfa">

## How to review this pull request
<!-- Provide step-by-step instructions to functionally test this PR. -->
<!-- Ensure that your code passing the repo's linting standards. -->
- [ ] Run `npm run develop`
- [ ] Open the Emulsify instance
- [ ] Click on the Breadcrumbs link in the sidebar
- [ ] Ensure that the breadcrumb works as expected.

mndonx committed:
feat(breadcrumb): draft of breadcrumb web component

joetower reviewed (commented):
@mndonx This looks good to me, nice work! 

My only question is: should we add a `current` or `active` state? Maybe a `status` option or something which we could style to distinguish it as the current path item  - which would not be a linked item? In this case, item level 3 would get the current/active state.

mndonx committed:
build(breadcrumb-component): merging in main

mndonx committed:
fix(breadcrumb-component): including special theming for current item

mndonx commented:
@joetower Thanks Joe! I've added contingency for the "current" item so it is not a link and has a special class. That is ready for review. 

However, I still have a couple issues to resolve. 
- @cienvaras showed us how to include some common variables in a different file, but I totally forgot to write it down or record it, so I will need a refresher on how it is done since that was a few weeks ago now! I'd like to include that here. 
- The colors are not passing contrast tests, but I don't want to mess with the variables. What is the process for changing the tokens?

mndonx committed:
fix(breadcrumb): updating with reusable classes

mndonx commented:
@joetower @cienvaras I've updated this PR to have a way to use a reusable class (`visually-hidden`.) Let me know what you think of the solution (I read about it in the Lit documentation.) 

There is definitely still an issue of color contrast. I added [an issue in the Emulsify UI Kit ](https://github.com/emulsify-ds/emulsify-ui-kit/issues/113)to address the contrast issues across the project. I wonder if I should wait on that, or move this forward?
